README.md

```diff
@@ -1,59 +1,380 @@
+    dyn_file.update("Number of houses", params.number_of_houses)
+    dyn_file.update("Number of floors", params.number_of_floors)
+    dyn_file.update("Depth", params.depth)
+    dyn_file.update("Width", params.width)
+    dyn_file.update("Height floor", params.height_floor)
+    dyn_file.update("Height roof", params.height_roof)
+
+    # generate updated file
+    input_file = dyn_file.generate()
+
+    return input_file, dyn_file
+```
+Let us go through the above mentioned logic:
+1. Retrieve the input files for the analysis, in this case the `dynamo_model_sample_app.dyn` file, and create a DynamoFile object instantiated from the `dynamo_model_sample_app.dyn` file.
+2. With the update method, the value of input nodes can be updated.
+3. When all inputs have been updated as desired, the generate method can be used to generate an updated `File` object.
+
+### Step 1.4 Define the app logic to convert the output.xml to a data_group
+In this step, we will define code to extract data from the dynamo file and output file. The dynamo file is used to get the node id, and the output.xml file is used to get the values. We will be creating another `staticmethod`  in the `controller`    class. See code below:
+
+
+```python
+from pathlib import Path
+from viktor import File
+from viktor.external.dynamo import DynamoFile
+
+@staticmethod
+def convert_dynamo_file_to_data_items(input_file: DynamoFile, output_file: File) -> DataGroup:
+    """Extracts the output of the Dynamo results by using the input and output files."""
+    # Collect ids for the computational output from the dynamo file (numerical output)
+    output_id_floor_area = input_file.get_node_id("(OUTPUT) Floor area per house")
+    output_id_total_cost = input_file.get_node_id("(OUTPUT) Total cost")
+    output_id_mki = input_file.get_node_id("(OUTPUT) MKI")
+    output_id_co2 = input_file.get_node_id("(OUTPUT) CO2")
+
+    # Collect the numerical results from the output file using the collected ids
+    with output_file.open_binary() as f:
+        floor_area = get_dynamo_result(f, id_=output_id_floor_area)
+        total_cost = get_dynamo_result(f, id_=output_id_total_cost)
+        mki = get_dynamo_result(f, id_=output_id_mki)
+        co2 = get_dynamo_result(f, id_=output_id_co2)
+
+    # Add values to a structured data group
+    data_group = DataGroup(
+        DataItem(label="Floor area", value=round(float(floor_area), 2), suffix="m²"),
+        DataItem(label="Total cost", value=round(float(total_cost), 2), suffix="€"),
+        DataItem(label="MKI", value=round(float(mki), 2)),
+        DataItem(label="CO₂ emission", value=round(float(co2), 2), suffix="ton CO₂"),
+    )
+    return data_group
+```
+Let us go through the above mentioned logic:
+1. Get the node_id, which corresponds to the same node id as the input file.
+2. Collect the numerical results.
+3. We create a structured data group from the collected numerical results using the `DataGroup` and `DataItem` classes.
```

Comment:
I'm not sure if the reader will understand what we mean by a structured DataGroup. Maybe good to mention that for more information on this, refer to:

https://docs.viktor.ai/docs/create-apps/results-and-visualizations/data-and-tables